packages/gatsby-link/src/index.js

```diff
@@ -149,6 +149,7 @@ class GatsbyLink extends React.Component {
                 // This is just a normal link to the current page so let's emulate default
                 // browser behavior by scrolling now to the top of the page.
                 window.scrollTo(0, 0)
+                document.querySelector(`#___gatsby`).focus()
```

Comment:
I believe you'll need to add a `tabindex="-1"` to the element, otherwise you cannot trigger programmatic focus.

This'll then likely show a focus outline that is not useful for most users, so you'd want to add some styles to remove this outline.